sui_programmability/examples/games/sources/vdf_based_lottery.move

```diff
@@ -0,0 +1,154 @@
+///  1) At least one player contributes true randomness,
+///  2) The number of iterations is defined such that it takes at least `submission_phase_length` to compute the VDF.
+module games::vdf_based_lottery {
+    use games::drand_lib::safe_selection;
+    use sui::clock::Clock;
+    use std::hash::sha2_256;
+    use sui::vdf::{hash_to_input, vdf_verify};
+
+    /// Error codes
+    const EGameNotInProgress: u64 = 0;
+    const EGameAlreadyCompleted: u64 = 1;
+    const EInvalidTicket: u64 = 3;
+    const ESubmissionPhaseInProgress: u64 = 4;
+    const EInvalidVdfProof: u64 = 5;
+    const ESubmissionPhaseFinished: u64 = 6;
+
+    /// Game status
+    const IN_PROGRESS: u8 = 0;
+    const COMPLETED: u8 = 2;
+
+    /// Game represents a set of parameters of a single game.
+    /// This game can be extended to require ticket purchase, reward winners, etc.
+    ///
+    public struct Game has key, store {
+        id: UID,
+        iterations: u64,
+        status: u8,
+        timestamp_start: u64,
+        submission_phase_length: u64,
+        participants: u64,
+        vdf_input_seed: vector<u8>,
+        winner: Option<u64>,
+    }
+
+    /// Ticket represents a participant in a single game.
+    /// Can be deconstructed only by the owner.
+    public struct Ticket has key, store {
+        id: UID,
+        game_id: ID,
+        participant_index: u64,
+    }
+
+    /// GameWinner represents a participant that won in a specific game.
+    /// Can be deconstructed only by the owner.
+    public struct GameWinner has key, store {
+        id: UID,
+        game_id: ID,
+    }
+
+    /// Create a shared-object Game.
+    public fun create(iterations: u64, submission_phase_length: u64, clock: &Clock, ctx: &mut TxContext) {
+        let game = Game {
+            id: object::new(ctx),
+            iterations,
+            status: IN_PROGRESS,
+            timestamp_start: clock.timestamp_ms(),
+            submission_phase_length,
+            vdf_input_seed: vector::empty<u8>(),
+            participants: 0,
+            winner: option::none(),
+        };
+        transfer::public_share_object(game);
+    }
+
+    /// Complete this lottery by sending VDF output and proof for the seed created from the
+    /// contributed randomness. Anyone can call this.
+    public fun complete(self: &mut Game, vdf_output: vector<u8>, vdf_proof: vector<u8>, clock: &Clock) {
+        assert!(self.status != COMPLETED, EGameAlreadyCompleted);
+        assert!(clock.timestamp_ms() - self.timestamp_start >= self.submission_phase_length, ESubmissionPhaseInProgress);
+
+        // Hash combined randomness to vdf input
+        let vdf_input = hash_to_input(&self.vdf_input_seed);
+        
+        // Verify output and proof
+        assert!(vdf_verify(&vdf_input, &vdf_output, &vdf_proof, self.iterations), EInvalidVdfProof);
+
+        // The randomness is derived from the VDF output by passing it through sha2_256 to make it uniform.
```

Comment:
sha256 was selected based on some rfc, why not blake? I'm ok, let's ensure we document exactly why.

Reply:
There's no RFC so we can definitely use blake instead.